apps/system/test/unit/browser_test.js

```diff
@@ -81,4 +92,26 @@ suite('system/Browser', function() {
     var app = MockAppWindowHelper.mLatest;
     assert.equal(app.isPrivate, true);
   });
+
+  test('when private browsing is the default setting', function() {
+    MockNavigatorSettings.mTriggerObservers('browser.private.default',
+        {settingValue: true});
+
+    setTimeout(function () {
```

Comment:
I have not yet, but did see this:

https://github.com/mozilla-b2g/gaia/blob/master/shared/test/unit/mocks/mock_navigator_moz_settings.js#L36-L37